cookbooks/aws-parallelcluster-config/resources/manage_efs.rb

```diff
@@ -14,16 +14,33 @@
 
 property :shared_dir_array, Array, required: true
 property :efs_fs_id_array, Array, required: true
+property :efs_encryption_in_transit_array, Array, required: true
+property :efs_iam_authorization_array, Array, required: true
 
 default_action :mount
 
 action :mount do
   efs_shared_dir_array = new_resource.shared_dir_array.dup
   efs_fs_id_array = new_resource.efs_fs_id_array.dup
-  efs_fs_id_array.zip(efs_shared_dir_array).each do |efs_fs_id, efs_shared_dir|
+  efs_encryption_in_transit_array = new_resource.efs_encryption_in_transit_array.dup
+  efs_iam_authorization_array = new_resource.efs_iam_authorization_array.dup
+
+  efs_fs_id_array.each_with_index do |efs_fs_id, index|
+    efs_shared_dir = efs_shared_dir_array[index]
+    efs_encryption_in_transit = efs_encryption_in_transit_array[index]
+    efs_iam_authorization = efs_iam_authorization_array[index]
+
     # Path needs to be fully qualified, for example "shared/temp" becomes "/shared/temp"
     efs_shared_dir = "/#{efs_shared_dir}" unless efs_shared_dir.start_with?('/')
 
+    # See reference of mount options: https://docs.aws.amazon.com/efs/latest/ug/automount-with-efs-mount-helper.html
+    mount_options = "_netdev,noresvport"
```

Comment:
Minor on code style: what about having mount options as an array and joining them ad the end?
This way we would avoid having commas here and there.

Reply:
This is not a minor note, let's avoid this kind of manipulation in future. Thanks for pointing this out.